compiler/rustc_expand/src/mbe/quoted.rs

```diff
@@ -81,14 +82,13 @@ pub(super) fn parse(
                                         })
                                         .unwrap_or_else(
                                             || {
-                                                let msg = format!(
-                                                    "invalid fragment specifier `{}`",
-                                                    frag.name
+                                                sess.dcx().emit_err(
+                                                    errors::InvalidFragmentSpecifier {
+                                                        span,
+                                                        name: frag.name.to_string(),
```

Comment:
Could you pass `Ident` directly here (and ofc update `InvalidFragmentSpecifier`)?

```suggestion
                                                        name: frag,
```

This way, we would properly print `` invalid fragment specifier `r#async` `` instead of `` invalid fragment specifier `async`  `` for:

```rs
macro_rules! m { ($x:r#async) => {} }
```